taqueria-vscode-extension/src/extension.ts

```diff
@@ -19,11 +19,14 @@ const { clearConfigWatchers, getConfigWatchers, addConfigWatcherIfNotExists } =
 
 	const getConfigWatchers = () => inMemoryState.configWatchers.values();
 
-	const addConfigWatcherIfNotExists = (folder: string, factory: () => api.FileSystemWatcher) => {
+	const addConfigWatcherIfNotExists = (folder: string, factory: () => api.FileSystemWatcher | null) => {
 		if (inMemoryState.configWatchers.has(folder)) {
 			return;
 		}
 		const watcher = factory();
+		if (watcher == null) {
```

Comment:
Is there a way we could avoid returning null here? Intuitively, I would think that a factory would either return a new object or throw.